### PR TITLE
Get EELS edges from a given energy window

### PR DESCRIPTION
**Pull request by ptim0626**
### Description of the change
A method to get EELS edges from a given energy window, similar to the one in EDS (get_xray_lines_near_energy). 

### Progress of the PR
- [x] Change implemented (can be split into several points),
- [x] update docstring (if appropriate),
- [x] update user guide (if appropriate),
- [x] add tests,
- [x] ready for review.

### Minimal example of the new feature
```python
>>> from hyperspy import signals
>>> s = signals.EELSSpectrum(np.ones(1024)) #dummy spectrum
>>> s.get_edges_near_energy(532)
['O_K']
>>>  s.get_edges_near_energy(532, width=10)
['O_K', 'Pd_M3', 'Sb_M5', 'Sb_M4']
```

### CODE REVIEW

File: hyperspy/_signals/eels.py [hunk 1]
```diff
@@ -154,6 +154,50 @@ def generate_subshells(self, include_pre_edges=False):
                                 '%s_%s' % (element, shell))
                             e_shells.append(subshell)
 
+    def get_edges_near_energy(self, energy, width=1):
```
Comment (ericpre) [inline hyperspy/_signals/eels.py]: 1 eV is fairly narrow, would 5 or 10 be a more sensible default width?

**Review by ericpre — COMMENTED** [review]
Nice and clean PR! I have one comments, otherwise it looks good to me.

**Comment by ericpre** [comment]
Github/Azure seems to have had hiccup and doesn't want to restart but since the last commit was only changes in the user guide, it is fine.

File: hyperspy/_signals/eels.py (hunk 1 above)
Comment (ptim0626) [reply hyperspy/_signals/eels.py]: Yes sure, will set the default to 10.

File: doc/user_guide/eels.rst [hunk 2]
```diff
@@ -33,6 +33,21 @@ information is stored in the :py:attr:`~.signal.BaseSignal.metadata`
 attribute (see :ref:`metadata_structure`). This information is saved to file
 when saving in the hspy format (HyperSpy's HDF5 specification).
 
+A method :py:meth:`~._signals.eels.EELSSpectrum_mixin.get_edges_near_energy`
+can be helpful to identify possible elements in the sample.
+:py:meth:`~._signals.eels.EELSSpectrum_mixin.get_edges_near_energy` returns a
+list of edges arranged in the order closest to the specified energy within a
+window.
```
Comment (tjof2) [inline doc/user_guide/eels.rst]: Best to define the window here, e.g. that you can control it with the width, and it's measured in eV

Comment (ptim0626) [reply doc/user_guide/eels.rst]: Good idea, I will elaborate a bit.

**Review by tjof2 — APPROVED** [review]

**Review by ericpre — APPROVED** [review]

**Comment by ericpre** [comment]
Thanks @ptim0626!

**Pull request merged**